extensions/interactions/ImageClickInput/directives/ImageClickInput.js

```diff
@@ -45,11 +45,41 @@ oppia.directive('oppiaInteractiveImageClickInput', [
             ($attrs.highlightRegionsOnHoverWithValue === 'true');
           $scope.filepath = imageAndRegions.imagePath;
           $scope.imageUrl = '';
-
-          ImagePreloaderService.getImageUrl($scope.filepath)
-            .then(function(objectUrl) {
-              $scope.imageUrl = objectUrl;
-            });
+          $scope.loadingIndicatorUrl = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/activity/loadingIndicator.gif')
+          );
+          $scope.isLoadingIndicatorShown = false;
+          if (ImagePreloaderService.inExplorationPlayer()) {
+            $scope.isLoadingIndicatorShown = true;
+            $scope.dimensions = (
+              ImagePreloaderService.getDimensionsOfImage($scope.filepath.name));
+            // For aligning the gif to the center of it's container
+            var loadingIndicatorSize = 120;
+            if ($scope.dimensions.height < 124) {
+              loadingIndicatorSize = 24;
+            }
+            var paddingTop = Math.max(0, (($scope.dimensions.height * 0.5) -
+              (loadingIndicatorSize * 0.5)));
+            $scope.loadingIndicatorContainerStyle =
+            {
+              'background-color': 'rgba(224,242,241,1)',
+              'padding-top': paddingTop + 'px',
+              height: $scope.dimensions.height + 'px'
+            };
+            $scope.loadingIndicatorStyle = {
+              display: 'block',
+              height: loadingIndicatorSize + 'px',
+              'margin-left': 'auto',
+              'margin-right': 'auto',
+              width: loadingIndicatorSize + 'px'
```

Comment:
Be consistent with the key names, use quotation marks on all keys.

Reply:
Then it gives linting errors that we don't need quotation marks for height and width.

Reply:
Hi @vojtechjelinek, we have a rule `quote-props` which is set as `as-needed`. Therefore the linter prompts to put quotes only when required otherwise there should be no quotes. (Refer to the [eslint documentation](https://eslint.org/docs/2.0.0/rules/quote-props#as-needed)).

Reply:
Thanks @apb7 .

Reply:
Okay, thanks. @ishucr7 Move things that are not dynamic (like `display: 'block'`) from here to CSS class in oppia.css and set it as class for the image. Do this for both css objects and in *ImageDirective.js* too.